iac-extensions/arm/src/main/java/org/sonar/iac/arm/parser/bicep/BicepGrammar.java

```diff
@@ -431,6 +432,7 @@ public TypeExpressionAble PRIMARY_TYPE_EXPRESSION() {
     return b.<TypeExpressionAble>nonterminal(BicepLexicalGrammar.PRIMARY_TYPE_EXPRESSION).is(
       b.firstOf(
         AMBIENT_TYPE_REFERENCE(),
+        ARRAY_TYPE_REFERENCE(),
```

Comment:
An idea to try to keep the capturing precedence here: move `literal_value` tothe top, then have `type_reference` with two alternatives (for now): ambient type reference and identifier.